.github/workflows/pull_request.yml

```diff
@@ -1,7 +1,13 @@
 name: Pipeline
 # consistency regarding formatting and idiomatic Rust
 
-on: [ pull_request ]
+on: 
+  push:
+    branches:
+      - dev
+  pull_request:
+    branches:
+      - "**"

```

Comment:
would this mean that the cash is used for every single branch? Even if we lets say realize that we have to reset the branch to an older state, would this mean that the cache would still be used for this PR?